File: parcial/src/main/java/co/unicauca/parcial/dto/StudentDTO.java
package co.unicauca.parcial.dto;


import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.PastOrPresent;
import lombok.*;

import java.util.Date;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class StudentDTO extends PersonDTO{

    @PastOrPresent(message = "{user.date.past}")
    private Date entryDate;
    @Email(message = "{email.constrain}")
    private String email;

    @Valid
    private AddressDTO address;

    @Valid
    Set<TelephoneDTO> telephones;
}

File: parcial-front/src/app/model/StudentDTO.ts
import { PersonDTO } from './PersonDTO';
import { AddressDTO } from './AdressDTO';
import { TelephoneDTO } from './TelephoneDTO';

export class StudentDTO extends PersonDTO {
    entryDate: Date;
    email: string;
    address: AddressDTO;
    telephones: TelephoneDTO[];

    constructor(
        idPerson: number,
        identificationNumber: string,
        identificationType: string,
        name: string,
        lastName: string,
        entryDate: Date,
        email: string,
        address: AddressDTO,
        telephones: TelephoneDTO[]
    ) {
        super(idPerson, identificationNumber, identificationType, name, lastName);
        this.entryDate = entryDate;
        this.email = email;
        this.address = address;
        this.telephones = telephones;
    }
}
